sklearn/datasets/kddcup99.py

```diff
@@ -0,0 +1,402 @@
+        f = BytesIO(urlopen(URL_).read())
+
+        dt = [('duration', int),
+              ('protocol_type', 'S4'),
+              ('service', 'S11'),
+              ('flag', 'S6'),
+              ('src_bytes', int),
+              ('dst_bytes', int),
+              ('land', int),
+              ('wrong_fragment', int),
+              ('urgent', int),
+              ('hot', int),
+              ('num_failed_logins', int),
+              ('logged_in', int),
+              ('num_compromised', int),
+              ('root_shell', int),
+              ('su_attempted', int),
+              ('num_root', int),
+              ('num_file_creations', int),
+              ('num_shells', int),
+              ('num_access_files', int),
+              ('num_outbound_cmds', int),
+              ('is_host_login', int),
+              ('is_guest_login', int),
+              ('count', int),
+              ('srv_count', int),
+              ('serror_rate', float),
+              ('srv_serror_rate', float),
+              ('rerror_rate', float),
+              ('srv_rerror_rate', float),
+              ('same_srv_rate', float),
+              ('diff_srv_rate', float),
+              ('srv_diff_host_rate', float),
+              ('dst_host_count', int),
+              ('dst_host_srv_count', int),
+              ('dst_host_same_srv_rate', float),
+              ('dst_host_diff_srv_rate', float),
+              ('dst_host_same_src_port_rate', float),
+              ('dst_host_srv_diff_host_rate', float),
+              ('dst_host_serror_rate', float),
+              ('dst_host_srv_serror_rate', float),
+              ('dst_host_rerror_rate', float),
+              ('dst_host_srv_rerror_rate', float),
+              ('labels', 'S16')]
+        DT = np.dtype(dt)
+
+        file = GzipFile(fileobj=f, mode='r')
+        Xy = []
+        for line in file.readlines():
+            Xy.append(line.replace('\n', '').split(','))
+        file.close()
+        print('extraction done')
+        Xy = np.asarray(Xy, dtype=object)
+        for j in range(42):
+            Xy[:, j] = Xy[:, j].astype(DT[j])
+
+        X = Xy[:, :-1]
+        y = Xy[:, -1]
+        # XXX bug when compress!=0:
+        # (error: 'Incorrect data length while decompressing[...] the file
+        #  could be corrupted.')
+
+        joblib.dump(X, samples_path, compress=0)
+        joblib.dump(y, targets_path, compress=0)
+
+    try:
+        X, y
+    except NameError:
+        X = joblib.load(samples_path)
+        y = joblib.load(targets_path)
+
+    if shuffle:
+        X, y = sh(X, y, random_state=random_state)
+
+    return Bunch(data=X, target=y, DESCR=__doc__)
+
+
+def _mkdirp(d):
+    """Ensure directory d exists (like mkdir -p on Unix)
+    No guarantee that the directory is writable.
+    """
+    try:
+        os.makedirs(d)
+    except OSError as e:
+        if e.errno != errno.EEXIST:
+            raise
+
+
+# XXX: to be moved (pb if moved in .base: circular importation with mldata)
```

Comment:
Go for it then


> On 6 août 2015, at 19:55, Nicolas Goix <notifications@github.com> wrote:
> 
> In sklearn/datasets/kddcup99.py:
> 
> > +
> > +    return Bunch(data=X, target=y, DESCR=__doc__)
> > +
> > +
> > +def _mkdirp(d):
> > +    """Ensure directory d exists (like mkdir -p on Unix)
> > +    No guarantee that the directory is writable.
> > +    """
> > +    try:
> > +        os.makedirs(d)
> > +    except OSError as e:
> > +        if e.errno != errno.EEXIST:
> > +            raise
> > +
> > +
> > +# XXX: to be moved (pb if moved in .base: circular importation with mldata)
> Yes we can and it is what is doing the function actually
> so it might be better just to remove it...
> 
> —
> Reply to this email directly or view it on GitHub.
>